### ISSUE TITLE
Investigate GitHub Actions artifact retention behavior

### ISSUE THREAD

**Issue by hknutsen** [equinor/ops-actions#431]
When using [actions/upload-artifact](https://github.com/actions/upload-artifact) to upload an artifact, you can set a custom retention using the `retention-days` input. For example, if you set this input to `7`, the artifact will be automatically deleted after 7 days.

Now consider a workflow with two jobs.

1. `build`: builds a .NET application and uploads it as an artifact with a retention of 1 day
2. `deploy`: downloads the artifact and deploys it to an Azure Web App

Let's imagine that the `deploy` job requires an approval. What if this job waits for an approval for more than 1 day? Will the artifact be deleted?

Basically, does the artifact retention period start as soon as the artifact is uploaded, or once the entire workflow is finished?
